### PR TITLE
corosync: 2.4.5 -> 3.1.2; kronosnet: init at 1.20

### PR DESCRIPTION
**Pull request by ryantm**
###### Motivation for this change

Trying to use corosync with pacemaker and want to have a recent version.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

closes #56804

### CODE REVIEW

**Comment by ryantm** [comment]
Result of `nixpkgs-review pr 119460` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>corosync</li>
    <li>kronosnet</li>
  </ul>
</details>

File: pkgs/development/libraries/kronosnet/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchFromGitHub
+, autoreconfHook, pkgconfig
+, libqb, libxml2, libnl, lksctp-tools
+, nss, openssl, bzip2, lzo, lz4, xz, zlib, zstd
+, doxygen
+}:
+
+with lib;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kronosnet/default.nix:8]:
```suggestion
```

File: pkgs/development/libraries/kronosnet/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig doxygen ];
+
+  buildInputs = [
+    libqb libxml2 libnl lksctp-tools
+    nss openssl
+    bzip2 lzo lz4 xz zlib zstd
+  ];
+
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kronosnet/default.nix:28]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/libraries/kronosnet/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    description = "VPN on steroids";
+    homepage = https://kronosnet.org/;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kronosnet/default.nix:30]:
```suggestion
    homepage = "https://kronosnet.org/";
```

File: pkgs/development/libraries/kronosnet/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    license = [ licenses.lgpl21Plus licenses.gpl2Plus ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kronosnet/default.nix:31]:
```suggestion
    license = with licenses; [ lgpl21Plus gpl2Plus ];
```

File: pkgs/development/libraries/kronosnet/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  pname = "kronosnet";
+  version = "1.20";
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-lP5W+4b9McU2Uqibh2SucIu2y4KluO3B1RpAJKgYq/M=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig doxygen ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kronosnet/default.nix:20]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config doxygen ];
```

File: pkgs/development/libraries/kronosnet/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+, nss, openssl, bzip2, lzo, lz4, xz, zlib, zstd
+, doxygen
+}:
+
+with lib;
+
+stdenv.mkDerivation rec {
+  pname = "kronosnet";
+  version = "1.20";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kronosnet/default.nix:13]:
```suggestion
  version = "1.20";

  src = fetchFromGitHub {
```

File: pkgs/development/libraries/kronosnet/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchFromGitHub
+, autoreconfHook, pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kronosnet/default.nix:2]:
```suggestion
, autoreconfHook, pkg-config
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by ryantm** [comment]
Result of `nixpkgs-review pr 119460` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>corosync</li>
    <li>kronosnet</li>
  </ul>
</details>

**Comment by ryantm** [comment]
@SuperSandro2000 Thanks for the review; I made all your suggested changes and made sure nixpkgs-hammering returns no errors for these.

**Pull request merged**